File: backend/models/task.go
package models

import (
	"time"
)

type Task struct {
	ID          uint      `gorm:"primaryKey" json:"id"`
	Title       string    `gorm:"not null" json:"title"`
	Description string    `json:"description"`
	DueDate     time.Time `json:"due_date"`
	Status      string    `gorm:"not null" json:"status"`
	Priority    string    `gorm:"not null" json:"priority"`
	UserID      uint      `gorm:"not null" json:"user_id"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

File: src/types/task.ts
export interface Task {
    id: string;
    title: string;
    description: string;
    status: 'todo' | 'in-progress' | 'completed';
    priority: 'low' | 'medium' | 'high';
    dueDate: string;
    tags: string[];
    createdAt: string;
    updatedAt: string;
}

export type TaskStatus = Task['status'];
export type TaskPriority = Task['priority'];

export interface TaskFormData {
    title: string;
    description: string;
    priority: TaskPriority;
    dueDate: string;
    tags: string[];
} 